src/library/scala/collection/immutable/Queue.scala

```diff
@@ -118,7 +118,9 @@ sealed class Queue[+A] protected(protected val in: List[A], protected val out: L
     *
     *  @param  elem        the element to insert
     */
-  def enqueue[B >: A](elem: B): Queue[B] = new Queue(elem :: in, out)
+  def enqueue[B >: A](elem: B): Queue[B] =
+    if (isEmpty) new Queue(Nil, elem :: Nil)
+    else new Queue(elem :: in, out)
```

Comment:
This seems like an easy win. I wonder if there is a good way to test it. Maybe a benchmark is the only test that really matters. If a test that just continually enqueues and dequeues a single element at a time doesn't get a bit faster, would anything?